docs/concepts/internationalization.md

```diff
@@ -0,0 +1,72 @@
+
+    A common pattern is using `getUnlocalizedName().substring(5)` to assign registry names. This is fragile and uses the unlocalized name for logic, which is considered bad practice. It could instead be reversed so that the registry name is set first, then the unlocalized name is set according to the registry name `MODID + "." + getRegistryName().getResourcePath()`.
+
+Localization methods
+--------------------
+
+!!! warning
+    A common issue is having the server localize for clients. The server can only localize in its own locale, which does not necessarily match the locale of connected clients.
+    
+    To respect the language settings of clients, the server should have clients localize text in their own locale using TextComponentTranslation or other methods preserving the language neutral translation keys.
+
+### `net.minecraft.client.resources.I18n` (client only)
+
+**This I18n class can only be found on a Minecraft client!** It is intended to be used by code that only runs on the client. Attempts to use this on a server will throw exceptions and crash.
+
+- `format(String, Object...)` localizes in the client's locale with formatting. The first parameter is a language key, and the rest are used for `String.format(String, Object...)`.
+
+### `net.minecraft.util.text.translation.I18n` (deprecated)
+
+**This class is deprecated and should often be avoided.** It is intended to be used by code that runs on the logical server. Because localization should rarely happen on the server, other alternatives should be considered.
+
+- `translateToLocal(String)` localizes in the game's locale without formatting. The parameter is a language key.
+- `translateToLocalFormatted(String, Object...)` localizes in the game's locale with formatting. The first parameter is a language key, and the rest are used for `String.format(String, Object...)`.
+
+### `TextComponentTranslation`
+
+`TextComponentTranslation` is an `ITextComponent` that is localized and formatted lazily. It is very useful when sending messages to players because it will be automatically localized in their own locale.
+
+The first parameter of the `TextComponentTranslation(String, Object...)` constructor is a language key, and the rest are used for formatting. The only supported format specifiers are `%s` and `%1$s`, `%2$s`, `%3$s` etc. Formatting arguments may be other `ITextComponent`s that will be inserted into the resulting formatted text with all their attributes preserved.
+
+### `TextComponentHelper`
+
+- `createComponentTranslation(ICommandSender, String, Object...)` creates a localized and formatted `ITextComponent` depending on a command sender. The localization and formatting is done eagerly if the command sender is a vanilla client. If not, the localization and formatting is done lazily with a `TextComponentTranslation`. This is only useful if the server should allow vanilla clients to connect.
```

Comment:
Don't you mean if the *receiver* of the message is a vanilla client? `sender` is the name of the parameter but this corresponds to ICommandSender.